src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -180,6 +181,24 @@ suggestRemoveRedundantImport ParsedModule{pm_parsed_source = L _  HsModule{hsmod
         = [("Remove import", [TextEdit (extendToWholeLineIfPossible contents _range) ""])]
     | otherwise = []
 
+suggestDeleteTopBinding :: ParsedModule -> Diagnostic -> [(T.Text, [TextEdit])]
+suggestDeleteTopBinding ParsedModule{pm_parsed_source = L _ HsModule{hsmodDecls}} Diagnostic{_range=_range,..}
+-- Foo.hs:4:1: warning: [-Wunused-top-binds] Defined but not used: ‘f’
+    | Just [name] <- matchRegex _message ".*Defined but not used: ‘([^ ]+)’"
+    , (_character . _start) _range == 0
+    , let sameName = filter (\(L l b) -> isTopLevel l && matchesBindingName b (T.unpack name)) hsmodDecls
+            = [("Delete ‘" <> name <> "’", [TextEdit (nextLine . srcSpanToRange $ l) "" | (L l _) <- sameName])]
+    | otherwise = []
+    where
+      isTopLevel l = (_character . _start) (srcSpanToRange l) == 0
```

Comment:
You are basically duplicating this logic in line 188. How about we change this to accept a range and then we can use it for both cases (composing with `srcSpanToRange` at one call site).

Reply:
Indeed! There was a bug with infix bindings, so I've eradicated this line to fix that bug